frame/staking-rewards/src/lib.rs

```diff
@@ -541,12 +615,15 @@ pub mod pallet {
 			duration_preset: DurationSeconds,
 			keep_alive: bool,
 		) -> Result<Self::PositionId, DispatchError> {
+			// Review [Andy]
+			// Before anything make sure we validate
+			ensure!(amount > Zero::zero(), Error::<T>::StakeIsZero);
+			ensure!(duration_preset > Zero::zero(), Error::<T>::DurationInvalid);
```

Comment:
this should use `Validated` on the parameters instead of validating in the function body imo